src/cmd/singularity/cli/sif.go

```diff
@@ -0,0 +1,250 @@
+parttype: System
+content: LINUX
+---------------------------`
+	sifHeaderExample = `
+sif header hah.sif
+================ SIF Header ================
+launch: #!/usr/bin/env run-singularity
+
+magic: SIF_MAGIC
+version: 0
+arch: AMD64
+uuid: 2b88f62f-be4f-4143-8a7a-061c49a68249
+creation time: Fri May 25 17:23:04 2018
+modification time: Fri May 25 17:23:04 2018
+number of descriptors: 1
+start of descriptors in file: 120
+length of descriptors in file: 104
+start of data in file: 3328
+length of data in file: 1MB
+============================================`
+)
+
+// SifCmd represent the sif CLI cmd
+var SifCmd = &cobra.Command{
+	Use:  "sif",
+	Args: cobra.MinimumNArgs(1),
+	Run:  nil,
+}
+
+// SifCreate sif create cmd
+var SifCreate = &cobra.Command{
+	Use:     "create [option] <file>",
+	Short:   "Create a new sif file with input data objects",
+	Example: sifCreateExample,
+	Args:    cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var argc []string
+		argc = append(argc, "create")
+		if partfile != "" {
+			argc = append(argc, []string{"-P", partfile}...)
+		}
+		if fstype != "" {
+			argc = append(argc, []string{"-f", fstype}...)
+		}
+		if parttype != "" {
+			argc = append(argc, []string{"-p", parttype}...)
+		}
+		if content != "" {
+			argc = append(argc, []string{"-c", content}...)
+		}
+		if uuID != "" {
+			argc = append(argc, []string{"-c", uuID}...)
+		}
+		argc = append(argc, args...)
+		SifCmd := exec.Command(sif, argc...)
+		SifCmd.Stdout = os.Stdout
+		SifCmd.Stderr = os.Stderr
+
+		if err := SifCmd.Start(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+		if err := SifCmd.Wait(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+	},
+}
+
+// SifList sif list subcommand
+var SifList = &cobra.Command{
+	Use:     "list <file>",
+	Short:   "List SIF data descriptors from an input SIF file",
+	Example: sifListExample,
+	Args:    cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		argc := []string{"list", args[0]}
+		SifCmd := exec.Command(sif, argc...)
+		SifCmd.Stdout = os.Stdout
+		SifCmd.Stderr = os.Stderr
+
+		if err := SifCmd.Start(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+		if err := SifCmd.Wait(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+	},
+}
+
+// SifInfo sif info subcommand
+var SifInfo = &cobra.Command{
+	Use:     "info [id] <file>",
+	Short:   "Print data object descriptor info",
+	Example: sifInfoExample,
+	Args:    cobra.MinimumNArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		argc := append([]string{"info"}, args...)
+		SifCmd := exec.Command(sif, argc...)
+		SifCmd.Stdout = os.Stdout
+		SifCmd.Stderr = os.Stderr
+
+		if err := SifCmd.Start(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+		if err := SifCmd.Wait(); err != nil {
+			sylog.Fatalf("%v", err)
+		}
+	},
+}
+
+// SifDump sif dump subcommand
+var SifDump = &cobra.Command{
+	Use:     "dump [id] <file>",
+	Short:   "Display data object content",
+	Example: sifInfoExample,
```

Comment:
The example says to expect the same output as `singularity sif info`, but this is not the case. The contents for the file are dumped to `stdout` instead.